posts/bayesian-model-comparison/index.qmd

```diff
@@ -184,73 +184,78 @@ where $\hat{\delta} = \frac{\Sigma_{i = 1}^{n} \delta_{i}}{n}$ and $\hat{\sigma}
 ```{python}
 import matplotlib.pyplot as plt
 import numpy as np
-import scipy
 from scipy import stats
-from sklearn.datasets import load_breast_cancer
+from sklearn.datasets import make_moons
 from sklearn.model_selection import RepeatedStratifiedKFold
-from sklearn.svm import SVC
+from sklearn.neural_network import MLPClassifier
 from sklearn.ensemble import RandomForestClassifier
 from sklearn.metrics import f1_score
 ```
 
 To see this in action the first thing we need is some data and repeated cross-validation predictions for two different algorithms.
-We use the *breast cancer* dataset from *sklearn*:
+We use sklearn to make a *moons* dataset:
 
 ```{python}
-data = load_breast_cancer()
-target = data["target"]
+X, y = make_moons(n_samples=600, noise=0.4, random_state=23)
 ```
 
-```{python}
-#| echo: false
-print(
-  f"Number of instances in total = {len(target)}"
-  f"\nOf which num_pos = {len(np.where(target==1)[0])} and "
-  f"num_neg = {len(np.where(target == 0)[0])}"
-)
-```
+This has an equal number of labels of each class.
 
-```{python}
-target_names = ["malignant", "benign"]
-X_cancer, Y_cancer = data.data, data.target
-```
-
-Now that we have the data let's run the repeated cross-validation for two models, here we use an SVM with linear kernel and a Random Forest classifier:
+Now that we have the data, let's run the repeated cross-validation for two models. Here we use a Random Forest and a neural network (MLPClassifier) classifier:
 
 ```{python}
+from time import time_ns
+
 n_repetitions = 10
 n_folds = 5
 
 RSKF = RepeatedStratifiedKFold(
   n_splits=n_folds, n_repeats=n_repetitions, random_state=23
 )
 
-models = {"svm": SVC, "rf": RandomForestClassifier}
+models = {
```

Comment:
I would actually still call them the actual models, but save them as model1 and model 2 variables,
i.e. model1 = "rf", model2="mlp".

Then propagate the vars instead of the names.

Reply:
i wasnt sure what you meant, but was other comment below. 

if we're done, and therefore know we will be using rf and mlp, i would just hard code if you want to refer to "rf" and "mlp" everywhere. no need to parametrise.

should i drop all references to model 1 and 2, also in text?

Reply:
I think the parametrisation could still be useful to limit the amount of changes needed. It would mean that only the md cells have to be modified. It's also straightforward to do so I don't see the downside. If we wanted to parametrise the md cells also we could use the `md` functionality in jupyter but that would require some configuration to make it nice so I would avoid that.